partcad-service-json-rpc/src/partcad_service_json_rpc/core/session.py

```
@@ -0,0 +1,177 @@
+        Mirrors the legacy LSP server's ``load_partcad``: a reload resets
+        PartCAD's global state between package loads.
+        """
+        with self._load_lock:
+            self.partcad_ctx = None
+            if self.partcad is None:
+                self.partcad = importlib.import_module("partcad")
+            else:
+                try:
+                    self.partcad.fini()
+                    self.partcad.logging_ansi_terminal_fini()
+                except Exception as e:  # pylint: disable=broad-except
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)